src/main/java/com/example/airbnbapi/service/MediaService.java

```diff
@@ -14,30 +15,46 @@
 public class MediaService {
 
 
-
     @Autowired
     private RepositoryFactory<Media> repositoryFactory;
 
 
     public Media insertOrUpdateItem(MediaType type, Media item) {
+        try {
+            return repositoryFactory.getRepositoryByType(type).save(item);
+        } catch (RuntimeException e) {
 
-       return repositoryFactory.getRepositoryByType(type).save(item);
+            throw new InternalServerErrorException("Sorry, Couldn't establish connection to the database");
```

Comment:
Perhaps make this something generic like `DatabaseException`? The service layer shouldn't be concerned with HTTP statuses so by virtue of naming your exception type as `InternalServerErrorException`, you'll be introducing a mix of concerns - or 'knowledge bleed' - into the service.

Reply:
Perhaps also catch `e` within the exception too, so it can be logged within the global exception handler?